proposals/date_replace.json

```diff
@@ -0,0 +1,99 @@
+{
+    "id": "date_replace",
+    "summary": "Replaces a part of a date",
+    "description": "Replaces a part of a given date string with a new value based on the provided unit, allowing for easy replacing of dates.",
```

Comment:
@soxofaan  What if we opt for truncating in this case i.e. if hour is set to 27, it can be truncated to 23. Similarly, if  day is set to -40, it can be truncated to the first day of the month

Reply:
yes indeed, clipping to the respective bounds of the date/time components makes most sense in this case, I think.
The alternative is throwing an error, but I'm not sure if that is useful here

Reply:
Cool, updated it to factor that.